cscareerhub/megabot
# [feature] Implement pin by react restricted by role, basic pin listing, permission utils, and dedent util

### Purpose of pull request

**Related Issue**: resolves #27, resolves #28

- Implement pin by react restricted by role, basic listing functionality, and permission utils
- This also adds a simple feature flag to the client message event - it'll only fire if the env is not prod
- Also included: reorganization of salary bot and ama bot tests
- Dedent util function for indented ES6 strings

### Pull request checklist


### Testing instructions

1. `++pins list` should send Career and Programming category pins to DMs
2. Send a message and react with the pin emoji
3. Check that the message got pinned
4. Undo the pin react on that same message
5. Check that the message got unpinned
6. Check that salary bot still works
